cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/AlertaSelfHealingNotifier.java

```diff
@@ -0,0 +1,282 @@
+            alertaMessage.setValue(logdir);
+            alertaMessages.add(alertaMessage);
+          });
+        });
+    }
+
+    private void alertMetricAnomaly(AnomalyType anomalyType, final String localHostname, DateTimeFormatter outFormatter,
+            List<AlertaMessage> alertaMessages, KafkaMetricAnomaly metricAnomaly) {
+        Map<BrokerEntity, Long> brokersEntitiesValues = metricAnomaly.entities();
+        for (Entry<BrokerEntity, Long> entry : brokersEntitiesValues.entrySet()) {
+          AlertaMessage alertaMessage = new AlertaMessage(localHostname + " - Broker " + entry.getKey().brokerId(),
+              anomalyType.toString());
+          alertaMessage.setSeverity(severityFromPriority(anomalyType.priority()));
+          alertaMessage.setGroup(AlertaAlertGroup.Performance.toString());
+          alertaMessage.setCreateTime(outFormatter.format(Instant.ofEpochMilli(entry.getValue())));
+          alertaMessages.add(alertaMessage);
+        }
+    }
+
+    private void alertBrokerFailure(AnomalyType anomalyType, final String localHostname, DateTimeFormatter outFormatter,
+            List<AlertaMessage> alertaMessages, BrokerFailures brokerFailures) {
+        Map<Integer, Long> failedBrokers = brokerFailures.failedBrokers();
+        for (Entry<Integer, Long> entry : failedBrokers.entrySet()) {
+          AlertaMessage alertaMessage = new AlertaMessage(localHostname + " - Broker " + entry.getKey(),
+              anomalyType.toString());
+          alertaMessage.setSeverity(severityFromPriority(anomalyType.priority()));
+          alertaMessage.setGroup(AlertaAlertGroup.Performance.toString());
+          alertaMessage.setCreateTime(outFormatter.format(Instant.ofEpochMilli(entry.getValue())));
+          alertaMessages.add(alertaMessage);
+        }
+    }
+
+    private void alertGoalViolation(AnomalyType anomalyType, final String localHostname, DateTimeFormatter outFormatter,
+            List<AlertaMessage> alertaMessages, GoalViolations goalViolations) {
+        Map<Boolean, List<String>> violations = goalViolations.violatedGoalsByFixability();
+        for (Entry<Boolean, List<String>> entry : violations.entrySet()) {
+          entry.getValue().forEach(goal -> { 
+            AlertaMessage alertaMessage = new AlertaMessage(localHostname, anomalyType.toString() + " - " + goal);
+            alertaMessage.setSeverity(severityFromPriority(anomalyType.priority()));
+            alertaMessage.setGroup(AlertaAlertGroup.Performance.toString());
+            alertaMessage.setCreateTime(outFormatter.format(Instant.ofEpochMilli(goalViolations.detectionTimeMs())));
+            alertaMessages.add(alertaMessage);
+          });
+        }
+    }
+
+    protected void sendAlertaMessage(AlertaMessage alertaMessage) throws IOException {
+      NotifierUtils.sendMessage(alertaMessage.toString(), _alertaApiUrl + "/alert", _alertaApiKey);
+    }
+
+    protected String severityFromPriority(int priority) {
```

Comment:
Nit: Should we 
1. make this function `static`,
2. make it get `AnomalyType anomalyType` as input rather than `int priority`
3. move it to `NotifierUtils` with a JavaDoc,
4. rename the `AlertaAlertSeverity` to something more generic (e.g. `AlertSeverity`),
5. rename the function to something that represents its new role (e.g. `getAlertSeverity`)
5. make this function return the enum rather than the String representation of it?

This sets a standard mapping between `AnomalyType` and their severity, and it makes it possible to reuse this function in other notifiers if needed.

Reply:
Ok I did it even if I'm not convinced at all as this function will return an enum which is in any case Alerta specific even if its name change. Indeed, `AlertSeverity.toString()` must return something which is directly used by the Alerta API otherwise we simply introduce a useless intermediate enum.